python_modules/dagster-graphql/dagster_graphql/implementation/fetch_resources.py

```diff
@@ -0,0 +1,48 @@
+import dagster._check as check
+from dagster._core.definitions.selector import RepositorySelector, ResourceSelector
+from dagster._core.host_representation.repository_location import RepositoryLocation
+from graphene import ResolveInfo
+
+from .utils import UserFacingGraphQLError, capture_error
+
+
+@capture_error
+def get_resources_or_error(graphene_info, repository_selector):
+    from ..schema.resources import GrapheneTopLevelResource, GrapheneTopLevelResources
+
+    check.inst_param(graphene_info, "graphene_info", ResolveInfo)
+    check.inst_param(repository_selector, "repository_selector", RepositorySelector)
+
+    location: RepositoryLocation = graphene_info.context.get_repository_location(
+        repository_selector.location_name
+    )
+    repository = location.get_repository(repository_selector.repository_name)
+    external_resources = repository.get_external_resources()
+
+    results = [
+        GrapheneTopLevelResource(external_resource) for external_resource in external_resources
+    ]
+
+    return GrapheneTopLevelResources(results=results)
+
```

Comment:
Why is this called top-level resource? Seems like it could be used to represent a resource anywhere in the tree?

Reply:
To contrast with the `GrapheneResource` which we return for e.g. job details. Will rename to `ResourceDetails` for now which should hopefully convey that it's a resource plus more display metadata for the resource page.